src/state/slices/validatorDataSlice/validatorDataSlice.test.ts

```diff
@@ -0,0 +1,104 @@
+import { clearState, store } from 'state/store'
+
+import { selectSingleValidator } from './selectors'
+import { validatorData } from './validatorDataSlice'
+
+const SHAPESHIFT_VALIDATOR_ADDRESS = 'cosmosvaloper199mlc7fr6ll5t54w7tts7f4s0cvnqgc59nmuxf'
+const SHAPESHIFT_OPPORTUNITY = {
+  address: SHAPESHIFT_VALIDATOR_ADDRESS,
+  moniker: 'Shapeshift DAO',
+  tokens: '111111',
+  apr: '0.24',
+  commission: '0.100000000000000000',
+}
+
+describe('validatorDataSlice', () => {
+  beforeEach(() => {
+    clearState()
+  })
+
+  it('returns uninitialized properties for initialState', async () => {
+    expect(store.getState().validatorData).toEqual({
+      byValidator: {},
+      validatorIds: [],
+    })
+  })
+
+  describe('reducers', () => {
+    describe('upsertValidatorData', () => {
+      it('updates or inserts validator data in state', async () => {
+        store.dispatch(
+          validatorData.actions.upsertValidatorData({
+            validators: [],
+          }),
+        )
+        expect(store.getState().validatorData.byValidator).toMatchSnapshot()
+      })
+    })
+
+    describe('clear', () => {
+      it('clears state back to initial state', async () => {
+        store.dispatch(
+          validatorData.actions.upsertValidatorData({
+            validators: [],
```

Comment:
nit : In order to verify that the clear is working well, it may be better to populate validator data with non empty content.

Reply:
Sure, great idea! At the end of the day empty array or an array with some data is the same in regards to clearing, but you're right it's probably a better idea to always have tests as close to the original data as possible.